### ISSUE TITLE
Room sort does not work as expected

### ISSUE THREAD

**Issue by Avi0n** [element-hq/element-x-ios#3269]
### Steps to reproduce

1. Tap People filter
2. See that the list is empty even though you have DMs
3. Clear filters and tap Rooms
4. See that DMs are included in the Rooms list
5. Clear filters and tap Favourites
6. See that the list is empty even though you have Favorites

#### What did you expect?
To see my DMs when tapping the People filter
#### What happened instead?
No DMs shown.

### Your phone model

iPhone 15 Pro Max

### Operating system version

iOS 18

### Application version

1.8.0 (122)

### Homeserver

matrix.org

### Will you send logs?

Yes

**Comment by Avi0n** [comment]
Forced use of Proxy in Dev settings and this behavior still exists. So maybe not related to SSS.

**Comment by pixlwave** [comment]
> Forced use of Proxy in Dev settings and this behavior still exists. So maybe not related to SSS.

Just to confirm, did you log out and back in again after changing the setting?

**Comment by Avi0n** [comment]
Yes, that's correct. I just submitted another set of logs with a screenshot attached.

**Comment by pixlwave** [comment]
Out of interest, are your DMs still shown in the People section on other apps like Element iOS or Element Web?

**Comment by Avi0n** [comment]
Yes, DMs show correctly with the People filter on Element iOS and Desktop.

**Comment by Avi0n** [comment]
FYI I tried this on EXA 0.5.3 (40005030) and had the exact same issue.

**Comment by Kimiblock** [comment]
Favorites are broken for me, people works fine

**Comment by stefanceriu** [comment]
@Avi0n is this ticket still relevant? 

I'm seeing quite a few `m.direct` errors in your logs

```
2024-09-11T17:37:12.661848Z  WARN matrix_sdk_base::client: Failed to deserialize a global account data event: leading sigil is incorrect or missing at line 1 column 2807 event_type="m.direct" | crates/matrix-sdk-base/src/client.rs:683 | spans: sync_once > handle_response > handle_account_data
```

It's supposed to be a dictionary of user identifiers to room indentifiers e.g.

```
{
  "type": "m.direct",
  "content": {
    "@user:homeserver.net" : ["!room:homeserver.net"]
    ...
  }
}
```

**Comment by stefanceriu** [comment]
Similar question for you @Kimiblock I guess, are your favorites working properly now? If not are there are interesting "tag" related warnings about them? 

Favorites are stored as tag on each room's account data e.g.

```
{
  "type": "m.tag",
  "content": {
    "tags": {
      "m.favourite": {}
    }
  }
}
```

**Comment by Kimiblock** [comment]
Yeah it is working now. I think it’s after migrating to SSS that things got fixed. Thanks!

**Comment by Avi0n** [comment]
> @Avi0n is this ticket still relevant? 

No, I fixed this by using /devtools on web and removing the malformed m.direct. Filters have been working as expected since then.

**Comment by stefanceriu** [comment]
Perfect, thank you!
